src/client/helpers/api-helper.js

```diff
@@ -31,31 +31,27 @@ export default class ApiHelper {
     }
   }
 
-  static validate(jsonString, version) {
-    return fetch(`/api/validate/${version}`, {
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      },
-      body: jsonString,
-    }).then(
-      res => ApiHelper.handleResponse(res, jsonString),
-    );
+  static validateJSON(jsonString, version, validationMode) {
+    return this.validate({ json: jsonString }, version, validationMode)
+      .then(res => ApiHelper.handleResponse(res, jsonString));
+  }
+
+  static validateURL(url, version, validationMode) {
+    return this.validate({ url }, version, validationMode)
+      .then(res => ApiHelper.handleResponse(res));
   }
 
-  static validateURL(url, version) {
-    return fetch(`/api/validateUrl/${version}`, {
+  static validate(data, version, validationMode) {
+    let body = JSON.parse(JSON.stringify(data));
```

Comment:
This seems fishy?

Reply:
This is something I was going to pick up with Joe during review, for some reason it's not liking the spread operator as I was hoping to just to `const body = { ...date, validationMode };` rather than clone the data object and add validationMode to it.

Reply:
If the spread isn’t working for whatever reason (weird it doesn’t) I noticed jQuery is in the project, it has it’s own method for that kind of thing https://api.jquery.com/jquery.extend/

TL;DR

```js
const body = $.extend({}, data, { validationMode });
```

Reply:
Thanks, Ben. Just seen this - we went for `Object.assign` which looks like it's the same thing.

Reply:
Nice! As long as that’s okay with browser support then that’s nice.

Basically no IE and in built Android browsers.

https://caniuse.com/#feat=mdn-javascript_builtins_object_assign

Reply:
Thanks for the warning, Ben. The site is a tool for devs so IE unlikely to be used.